src/main/java/com/iri/dao3/dao/jdbc/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+package com.iri.dao3.dao.jdbc;
+
+import com.iri.dao3.dao.CarDao;
+import com.iri.dao3.exception.DataProcessingException;
+import com.iri.dao3.lib.Dao;
+import com.iri.dao3.model.Car;
+import com.iri.dao3.model.Driver;
+import com.iri.dao3.model.Manufacturer;
+import com.iri.dao3.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        String updateQuery = "INSERT INTO cars (model, manufacturer_id) "
+                + "VALUES (?, ?) ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(updateQuery, Statement.RETURN_GENERATED_KEYS)
+        ) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert car in method create", e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectQuery = "SELECT c.id as id, c.model as model, "
+                + "m.id as manufacturer_id, m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturer m ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted != TRUE AND m.deleted !=TRUE AND c.id = ? ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)
+        ) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            Car car = null;
+            while (resultSet.next()) {
+                car = constructCarFromResultSet(resultSet).get();
+            }
+            car.setDrivers(getAllDriversByCar(car));
+            return Optional.of(car);
```

Comment:
```suggestion
            return Optional.ofNullable(car);
```
`Optional.of()` will throw NPE if car is not found, null-check should be handled by the service

Reply:
ok